src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -170,29 +220,35 @@ func (sm *StorageMinerActor_I) failSector(sectorNo sector.SectorNumber, incremen
 	case SectorActiveSN:
 		// wont be terminated from Active
 		sm.SectorTable().Impl().ActiveSectors_ -= 1
+		sm.SectorTable().Impl().FailingSectors_ += 1
+		sm.ProvingSet_.Remove(sectorNo)
+		sm.Sectors()[sectorNo].Impl().State_ = SectorFailing(newFaultCount)
 	case SectorCommittedSN:
 		sm.SectorTable().Impl().CommittedSectors_ -= 1
+		sm.SectorTable().Impl().FailingSectors_ += 1
+		sm.ProvingSet_.Remove(sectorNo)
+		sm.Sectors()[sectorNo].Impl().State_ = SectorFailing(newFaultCount)
 	case SectorRecoveringSN:
 		sm.SectorTable().Impl().RecoveringSectors_ -= 1
+		sm.SectorTable().Impl().FailingSectors_ += 1
+		sm.ProvingSet_.Remove(sectorNo)
+		sm.Sectors()[sectorNo].Impl().State_ = SectorFailing(newFaultCount)
 	case SectorFailingSN:
-
+		// no change to SectorTable but increase in FaultCount
+		sm.Sectors()[sectorNo].Impl().State_ = SectorFailing(newFaultCount)
 	default:
 		// TODO: proper failure
 		panic("Invalid sector state in CronAction")
 	}
 
-	sm.SectorTable().Impl().FailingSectors_ += 1
+	// TODO commit state change
 
 	if newFaultCount > MAX_CONSECUTIVE_FAULTS {
 		// TODO: heavy penalization: slash pledge collateral and delete sector
 		// TODO: SendMessage(SPA.SlashPledgeCollateral)
-		sm.SectorTable().Impl().TerminatedSectors_ += 1
-		sm.SectorTable().Impl().FailingSectors_ -= 1
-		sm.clearSector(sectorNo)
-	}
 
-	sm.ProvingSet_.Remove(sectorNo)
-	sm.Sectors()[sectorNo].Impl().State_ = SectorFailing(newFaultCount)
+		sm.clearSector(sectorNo, true)
```

Comment:
given that this only happens _once_ (in one code point, here) and that `sm.SectorTable().Impl().TerminationFaultCount_` is not a sector state, i think we should:

- move back to `clearSector(sectorNum)` (not taking in `isTerminated`) and just add the fault count here:

```
sm.clearSector(sectorNo)
sm.SectorTable().Impl().TerminationFaultCount_++
```